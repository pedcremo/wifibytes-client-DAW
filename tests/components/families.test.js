import Families from '../../src/components/catalog/families';
import familia from "../json_endpoints/familia.json";
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

const families = Enzyme.shallow(<Families familia={familia} />);

it('We can check if Families component called the class constructor', () => {
    expect(families).toMatchSnapshot();
});

it('Families render must be called and it works properly', () => {
    expect(families).toHaveLength(1);
});

it('We check if data has been printed', () => {
    expect(families.find('.display-3').getElement().props.children).toEqual([175, " â‚¬ IVA Inc."]);
    expect(families.find('.display-1').getElement().props.children).toEqual("Smartphones");
});