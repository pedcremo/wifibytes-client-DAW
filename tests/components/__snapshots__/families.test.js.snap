// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We can check if Families component called the class constructor 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Families />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="grid row"
        >
          <div
            className="col"
          >
            <h1
              className="text-center"
            />
            <h2
              className="text-center"
            />
          </div>
        </div>,
        <ul
          className="nav-families nav nav-pills mb-3"
          id="pills-tab"
          role="tablist"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col"
          >
            <h1
              className="text-center"
            />
            <h2
              className="text-center"
            />
          </div>,
          "className": "grid row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="text-center"
              />,
              <h2
                className="text-center"
              />,
            ],
            "className": "col",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "text-center",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "text-center",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "nav-families nav nav-pills mb-3",
          "id": "pills-tab",
          "role": "tablist",
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="grid row"
          >
            <div
              className="col"
            >
              <h1
                className="text-center"
              />
              <h2
                className="text-center"
              />
            </div>
          </div>,
          <ul
            className="nav-families nav nav-pills mb-3"
            id="pills-tab"
            role="tablist"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col"
            >
              <h1
                className="text-center"
              />
              <h2
                className="text-center"
              />
            </div>,
            "className": "grid row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="text-center"
                />,
                <h2
                  className="text-center"
                />,
              ],
              "className": "col",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "text-center",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "text-center",
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "nav-families nav nav-pills mb-3",
            "id": "pills-tab",
            "role": "tablist",
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
